File: frontend/src/api/portfolio.ts
import request from './request'
import type {
  Portfolio,
  PortfolioDetail,
  PortfolioListRes,
  CreatePortfolioReq,
  UpdatePortfolioReq,
  CreatePortfolioItemReq,
  UpdatePortfolioItemReq,
  SortPortfolioItemsReq,
} from '@/types/portfolio'

export const portfolioApi = {
  getList(params?: {
    page?: number
    page_size?: number
    keyword?: string
    status?: number
    category_id?: string
  }) {
    return request.get<PortfolioListRes>('/portfolios', { params })
  },
  getById(id: string) {
    return request.get<PortfolioDetail>(`/portfolios/${id}`)
  },
  create(data: CreatePortfolioReq) {
    return request.post<Portfolio>('/portfolios', data)
  },
  update(id: string, data: UpdatePortfolioReq) {
    return request.put<Portfolio>(`/portfolios/${id}`, data)
  },
  remove(id: string) {
    return request.delete(`/portfolios/${id}`)
  },
  addItem(id: string, data: CreatePortfolioItemReq) {
    return request.post(`/portfolios/${id}/items`, data)
  },
  updateItem(id: string, itemId: string, data: UpdatePortfolioItemReq) {
    return request.put(`/portfolios/${id}/items/${itemId}`, data)
  },
  removeItem(id: string, itemId: string) {
    return request.delete(`/portfolios/${id}/items/${itemId}`)
  },
  sortItems(id: string, data: SortPortfolioItemsReq) {
    return request.put(`/portfolios/${id}/items/sort`, data)
  },
}

File: backend/internal/router/portfolio.go
package router

import (
	"cus-cms/internal/controller"

	"github.com/gin-gonic/gin"
)

// RegisterPortfolioRoutes 注册摄影作品集管理路由。
func RegisterPortfolioRoutes(r *gin.RouterGroup, portfolioController *controller.PortfolioController, authMiddleware gin.HandlerFunc) {
	portfolios := r.Group("/portfolios")
	portfolios.Use(authMiddleware)
	{
		portfolios.POST("", portfolioController.Create)
		portfolios.GET("", portfolioController.GetList)
		portfolios.GET("/:id", portfolioController.GetByID)
		portfolios.PUT("/:id", portfolioController.Update)
		portfolios.DELETE("/:id", portfolioController.Delete)

		portfolios.POST("/:id/items", portfolioController.AddItem)
		// 注意：items/sort 必须在 items/:itemId 之前注册，否则会被 :itemId 匹配
		portfolios.PUT("/:id/items/sort", portfolioController.SortItems)
		portfolios.PUT("/:id/items/:itemId", portfolioController.UpdateItem)
		portfolios.DELETE("/:id/items/:itemId", portfolioController.DeleteItem)
	}
}
